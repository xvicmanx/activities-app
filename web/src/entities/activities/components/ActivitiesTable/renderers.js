// @flow

import React from 'react';

import CommunitiesDropdown from '../../../communities/components/CommunitiesDropdown';


type RendererProps = {
  field: {
    value: string | number,
    name: string,
    onChange: Function,
  },
};

export const DescriptionRenderer = ({ field }: RendererProps): React$Element<any> => (
  <textarea {...field} />
);

export const CommunitiesSelectRenderer = ({ field }: RendererProps): React$Element<any> => (
  <CommunitiesDropdown
    name={field.name}
    value={field.value}
    onChange={field.onChange}
  />
);
