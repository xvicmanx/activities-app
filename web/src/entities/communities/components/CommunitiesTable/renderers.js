// @flow

import React from 'react';

type RendererProps = {
  field: Object,
};

export const DescriptionRenderer = ({
  field,
}: RendererProps): React$Element<any> => <textarea {...field} />;
