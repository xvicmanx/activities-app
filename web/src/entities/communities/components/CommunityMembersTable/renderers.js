// @flow

import React from 'react';

import UsersDropdown from '../../../users/components/UsersDropdown';
import BooleanDropdown from '../BooleanDropdown';

type RendererProps = {
  field: Object,
};

export const MembersSelectRenderer = ({ field }: RendererProps): React$Element<any> => (
  <UsersDropdown {...field} />
);

export const CoordinatesSelectRenderer = ({ field }: RendererProps): React$Element<any> => (
  <BooleanDropdown {...field} />
);
