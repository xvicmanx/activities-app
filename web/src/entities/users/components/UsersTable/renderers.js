//@flow

import React from 'react';
import { CopyToClipboard } from 'react-copy-to-clipboard';

const styles = {
  copy: {
    backgroundColor: '#5caaf6',
    cursor: 'pointer',
    marginLeft: '10px',
    padding: '5px',
    borderRadius: '5px',
    color: '#fff',
  },
};

type RendererProps = {
  field: {
    value: string,
  },
};

export const DescriptionRenderer = ({
  field,
}: RendererProps): React$Element<any> => <textarea {...field} />;

export const PasswordRenderer = ({
  field,
}: RendererProps): React$Element<any> => (
  <div>
    <input {...field} type="password" />
    <CopyToClipboard text={field.value}>
      <span style={styles.copy}>Copiar</span>
    </CopyToClipboard>
  </div>
);
