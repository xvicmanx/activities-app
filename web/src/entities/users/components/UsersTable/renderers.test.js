import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import { DescriptionRenderer, PasswordRenderer } from './renderers';

describe('UsersTable renderers', () => {
  describe('DescriptionRenderer', () => {
    it('renders properly', () => {
      const renderer = new ShallowRenderer();
      const result = renderer.render(<DescriptionRenderer
        field={{
          name: 'test-name',
          value: 'test-value',
          onChange: jest.fn(),
        }}
      />);
      expect(result).toMatchSnapshot();
    });
  });

  describe('PasswordRenderer', () => {
    it('renders properly', () => {
      const renderer = new ShallowRenderer();
      const result = renderer.render(<PasswordRenderer
        field={{
          name: 'test-name',
          value: 'test-value',
          onChange: jest.fn(),
        }}
      />);
      expect(result).toMatchSnapshot();
    });
  });
});
