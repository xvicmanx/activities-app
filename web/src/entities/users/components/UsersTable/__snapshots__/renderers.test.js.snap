// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UsersTable renderers DescriptionRenderer renders properly 1`] = `
<textarea
  name="test-name"
  onChange={[MockFunction]}
  value="test-value"
/>
`;

exports[`UsersTable renderers PasswordRenderer renders properly 1`] = `
<div>
  <input
    name="test-name"
    onChange={[MockFunction]}
    type="password"
    value="test-value"
  />
  <CopyToClipboard
    text="test-value"
  >
    <span
      style={
        Object {
          "backgroundColor": "#5caaf6",
          "borderRadius": "5px",
          "color": "#fff",
          "cursor": "pointer",
          "marginLeft": "10px",
          "padding": "5px",
        }
      }
    >
      Copiar
    </span>
  </CopyToClipboard>
</div>
`;
